rtdb2/RtDB2Configuration.cpp

```diff
@@ -60,337 +28,73 @@ RtDB2Configuration::RtDB2Configuration() {
-                continue;
-            xercesc::DOMElement* element = dynamic_cast<xercesc::DOMElement*>(current_node);
-            std::string tag_name = xercesc::XMLString::transcode(element->getTagName());
-            if (tag_name != "key") {
-                std::stringstream ss;
-                ss << "Unknown tag \"";
-                ss << tag_name << "\" was found";
-                err_handler->create_error(CustomErrorHandler::WARNING, ss.str());
-                continue;
-            }
-
-            xercesc::DOMAttr* attr_id = element->getAttributeNode(name_id);
-            xercesc::DOMAttr* attr_shared = element->getAttributeNode(name_shared);
-            xercesc::DOMAttr* attr_period = element->getAttributeNode(name_period);
-            xercesc::DOMAttr* attr_phase = element->getAttributeNode(name_phase);
-            xercesc::DOMAttr* attr_timeout = element->getAttributeNode(name_timeout);
-            xercesc::DOMAttr* attr_oid = element->getAttributeNode(name_oid);
-
-            // Check if the attribute nome and shared exist
-            if (attr_id == NULL) {
-                err_handler->create_error(CustomErrorHandler::WARNING, "Skipping a key - missing attribute id or shared");
-                continue;
-            }
-            std::string id = to_upper(xercesc::XMLString::transcode(attr_id->getValue()));
-            if (attr_oid != NULL) {
-                std::string oid = xercesc::XMLString::transcode(attr_oid->getValue());
-                if (!is_number(oid)) {
-                    err_handler->create_error(CustomErrorHandler::WARNING, "oid from tag \"key\" is invalid");
-                } else {
-                    associate_keys_int_string(atoi(oid.c_str()), id);
+        compressor_settings_.use_dictionary = config->General().Compressor().dictionary();
+
+        // DefaultKeyValue
+        default_details_.period = config->General().DefaultKeyValue().period();
+        default_details_.phase_shift = config->General().DefaultKeyValue().phase();
+        default_details_.shared = config->General().DefaultKeyValue().shared();
+        default_details_.timeout = config->General().DefaultKeyValue().timeout();
+
+        // Keys
+        if (config->Keys().present()) {
+            for (rtdbconfig::Keys::key_const_iterator key (config->Keys().get().key().begin());
+                key != config->Keys().get().key().end(); ++key)
+            {
+                KeyDetail kd;
+                kd.period = key->period().present() ? key->period().get() : default_details_.period;
+                kd.phase_shift = key->phase().present() ? key->phase().get() : default_details_.phase_shift;
+                kd.shared = key->shared().present() ? key->shared().get() : default_details_.shared;
+                kd.timeout = key->timeout().present() ? key->timeout().get() : default_details_.timeout;
+                insert_key_detail(to_upper(key->id()), kd);
+
+                if (key->oid().present()) {
```

Comment:
The 'oid' element was removed in the falcons_2.5 version, because we thought it wouldn't hurt to drop legacy rtdb1. But maybe that was too soon? To be resolved in the other pull request.

Reply:
Agree. Let's take this as a separate subject, if needed.